### PR TITLE
[Draft PR; Do Not Merge] Dockerfile Testing

### PR DESCRIPTION
**Pull request by inf0rmatiker**
This Draft PR is only for comparing changes/updates to a new Dockerfile for the Lustre CSI driver. Ignore other configuration files or helper scripts that get pushed up to this branch; they will not be merged into master @roehrich-hpe

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,53 @@
-FROM golang:1.17 as builder
+FROM centos:centos8 AS base
+
+WORKDIR /
+
+# Install basic dependencies
```
Comment (inf0rmatiker) [inline Dockerfile:5]: Had to update repo mirrors here to get basic deps

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,53 @@
...
+
+#
+# Note: The COPY commands below have the potential to invalidate any layer
+# that follows.
+#
+
+FROM base as builder
+
+# Install go
+RUN wget https://go.dev/dl/go1.17.6.linux-amd64.tar.gz
```
[the remark this answers is not in the document]
Comment (NateRoiger) [reply Dockerfile:46]: Can't _builder_ stay the same and your application image is what pulls in all the RPMs? I don't see the point of _base_ here.

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,53 @@
-FROM golang:1.17 as builder
+FROM centos:centos8 AS base
+
+WORKDIR /
+
+# Install basic dependencies
+RUN sed -i -e "s|mirrorlist=|#mirrorlist=|g" /etc/yum.repos.d/CentOS-*
+RUN sed -i -e "s|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g" /etc/yum.repos.d/CentOS-*
+RUN dnf install -y make git gzip wget gcc tar
```
Comment (roehrich-hpe) [inline Dockerfile:8]: We probably need only wget  and tar here.  Will you  please verify that?

File: sendit.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (roehrich-hpe) [inline sendit.sh:1]: Don't put this file  in here.

File: Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (inf0rmatiker) [reply Dockerfile]: Ok I addressed this. Now we just have builder and a final application stage.

File: config/lustre/manager_volumes_patch.yaml [hunk 5]
```diff
@@ -22,6 +22,10 @@ spec:
             name: ulib-dir
           - mountPath: /usr/lib64
             name: lib64-dir
+          - mountPath: /bin
+            name: bin-dir
+          - mountPath: /sbin
+            name: sbin-dir
```
Comment (roehrich-hpe) [inline config/lustre/manager_volumes_patch.yaml:28]: Remove all of the mounts that pull in binaries and libraries.  We're installing that stuff from the rpm that you have in the Dockerfile.

**Pull request closed without merging**